src/shared/services/shortcut-action.service.ts

```diff
@@ -0,0 +1,232 @@
+              public domUtilService: DomUtilService,
+              public jsonStoreService: JsonStoreService,
+              public jsonSchemaService: JsonSchemaService,
+              public pathUtilService: PathUtilService,
+              public tabIndexService: TabIndexService) { }
+
+  addToRootAction(path: Array<any>): void {
+    this.add(path, true);
+  }
+
+  addAction(path: Array<any>): void {
+    this.add(path, false);
+  }
+
+  private add(path: Array<any>, root: boolean): void {
+    let _path = this.pathUtilService.getNearestOrRootArrayParentInPath(path, root);
+    this.addNewElementInArray(_path, this.jsonSchemaService.getSchemaFromPath(_path));
+  }
+
+  /**
+   * @param {Array<any>} path - Path of the array parent that the element is about to be inserted
+   * @param {Object} schema - Schema of the element that is about to be inserted
+   */
+  private addNewElementInArray(path: Array<any>, schema: Object) {
+    let itemSchema = schema['items'];
+    let emptyValue = this.emptyValueService.generateEmptyValue(itemSchema);
+    let values = this.jsonStoreService.getIn(path) || List();
+    this.jsonStoreService.setIn(path, values.push(emptyValue));
+    path.push(values.size);
+    setTimeout(() => {
+      this.updateTabIndexesAndFocusElementInPath(path.join('.'));
+    });
+  }
+
+  moveUpAction(path: Array<any>): void {
+    this.move(path, -1);
+  }
+
+  moveDownAction(path: Array<any>): void {
+    this.move(path, 1);
+  }
+
+  /**
+   * @param {Array<any>} path - Path of the element that is moved
+   * @param {number} direction - Movement direction. -1 for UP, +1 for DOWN
+   */
+  private move(path: Array<any>, direction: number): void {
+    let index = this.pathUtilService.getElementIndexInForwardOrReversePath(path, false);
+    path[path.length - 2] = this.moveElement(index, direction, this.pathUtilService.getNearestOrRootArrayParentInPath(path, false));
+    let pathString = path.join('.');
+    setTimeout(() => {
+      this.domUtilService.flashElementById(pathString);
+      this.domUtilService.focusAndSelectFirstInputChildById(pathString);
+    });
+  }
+
+  /**
+   * @param {number} index - Index of the element that is moved
+   * @param {number} direction - Movement direction. -1 for UP, +1 for DOWN
+   * @returns {number} - Returns the new index of the moved element
+   */
+  private moveElement(index: number, direction: number, path: Array<any>): number {
+    let values =  this.jsonStoreService.getIn(path);
+    let newIndex = ((index + direction) < values.size &&
+                    (index + direction) >= 0) ? index + direction : values.size - Math.abs((index + direction));
+    let temp = values.get(index);
+    values = values
+      .set(index, values.get(newIndex))
+      .set(newIndex, temp);
+    this.jsonStoreService.setIn(path, values);
+    return newIndex;
+  }
+
+  deleteAction(path: Array<any>) {
+    if (window.confirm('Are you sure you want to delete this item?')) {
```

Comment:
We can remove it since the button to Delete will now be hidden under a Dropdown and then there is undo, so an accident can be recovered.